### PR TITLE
feat: Add add metadata options to managed node groups launch template

### PR DESCRIPTION
**Pull request by xr0n1ck**
# PR o'clock

## Description

Please explain the changes you made here and link to any relevant issues.

### Checklist

- [x] README.md has been updated after any changes to variables and outputs. See https://github.com/terraform-aws-modules/terraform-aws-eks/#doc-generation

### CODE REVIEW

File: modules/node_groups/launch_template.tf [hunk 1]
```diff
@@ -63,6 +63,12 @@ resource "aws_launch_template" "workers" {
     ])
   }
 
+  metadata_options {
+    http_endpoint               = lookup(each.value, "metadata_http_endpoint", "enabled")
```
Comment (antonbabenko) [inline modules/node_groups/launch_template.tf:67]: Let's generate `metadata_options` using `dynamic` and create it only if one of 3 inputs are present. Exactly as here - https://github.com/terraform-aws-modules/terraform-aws-autoscaling/blob/master/main.tf#L92-L99

**Review by antonbabenko — CHANGES_REQUESTED** [review]

**Comment by daroga0002** [comment]
looks that it is duplicating https://github.com/terraform-aws-modules/terraform-aws-eks/pull/1485

**Comment by antonbabenko** [comment]
@xr0n1ck If you can verify it and apply the changes to that PR, it would be great.

**Comment by xr0n1ck** [comment]
@antonbabenko 
As far as I have no write permissions to that repo, I can create pr to that branch  and wait to approval.
It will be easier and faster to create new one, since dynamic metadata little bit different

**Comment by antonbabenko** [comment]
I understand, let's make a PR based on that one into this repository that will close #1485

**Comment by maxbrunet** [comment]
Hello! I have rebased #1485

**Comment by bryantbiggs** [comment]
hi @maxbrunet / @xr0n1ck - whichever PR we go with, can we just copy what we have for the autoscaling group module https://github.com/terraform-aws-modules/terraform-aws-autoscaling/blob/d4b7da041ef781604726953234b5bcd494cc5908/main.tf#L92-L99

you should be able to copy that block, and its variable and paste them in your PRs

**Comment by maxbrunet** [comment]
@bryantbiggs I copied what was already done in this module, I was thinking to be consistent with that:
https://github.com/terraform-aws-modules/terraform-aws-eks/blob/c2bd137152945317124c4cded258f12662d267f4/workers_launch_template.tf#L385-L401
https://github.com/terraform-aws-modules/terraform-aws-eks/blob/c2bd137152945317124c4cded258f12662d267f4/workers.tf#L268-L284

Also in the autoscaling module you shared, defaults being `null` adds to the inconsistency, and this errors if `http_endpoint` is not specified:
```hcl
http_endpoint               = lookup(metadata_options.value, "http_endpoint", null)
``` 
`http_endpoint` must be one of `enabled` or `disabled`.

**Comment by bryantbiggs** [comment]
I'm confused, what you said is inconsistent is what you have in your PR https://github.com/terraform-aws-modules/terraform-aws-eks/pull/1485/files#diff-1044253a991537ca26fb2e4d8ffd2689492a98061b0c1cada5531a90516fae71R79

**Comment by maxbrunet** [comment]
I apply the defaults here https://github.com/terraform-aws-modules/terraform-aws-eks/pull/1485/files#diff-98193ca5efaaf91be09ba7fc2c847ee1d2df56aa4e9d3691b7fa4e5fffc318edR29-R31

So in code, it is consistent with how the node_groups module applies defaults, and the behaviour is consistent with the self-managed workers

**Pull request closed without merging**